tests/foreman/longrun/test_oscap.py

```diff
@@ -269,6 +178,125 @@ def test_positive_upload_to_satellite(self):
+            make_oscappolicy(
+                session,
+                content=policy_values.get('content'),
+                host_group=policy_values.get('hgrp'),
+                name=policy_values.get('policy'),
+                period=OSCAP_PERIOD['weekly'],
+                profile=OSCAP_PROFILE['common'],
+                period_value=OSCAP_WEEKDAY['friday'],
+            )
+            # Creates two vm's each for rhel6 and rhel7, runs
+            # openscap scan and uploads report to satellite6.
+            with VirtualMachine(distro=vm_values.get('distro')) as vm:
+                host = vm.hostname
+                vm.install_katello_ca()
+                vm.register_contenthost(
+                    self.config_env['org_name'],
+                    self.config_env['ak_name']
+                )
+                vm.configure_puppet(vm_values.get('rhel_repo'))
+                self.hosts.update(
+                    name=vm._target_image,
+                    domain_name=vm._domain,
+                    parameters_list=[
+                        ['Host', 'Lifecycle Environment',
+                         self.config_env['env_name']],
+                        ['Host', 'Content View', self.config_env['cv_name']],
+                        ['Host', 'Host Group', vm_values.get('hgrp')],
+                        ['Host', 'Reset Puppet Environment', True],
+                    ],
+                )
+                session.nav.go_to_hosts()
+                # Run "puppet agent -t" twice so that it detects it's,
+                # satellite6 and fetch katello SSL certs.
+                for _ in range(2):
+                    vm.run(u'puppet agent -t 2> /dev/null')
+                result = vm.run(
+                    u'cat /etc/foreman_scap_client/config.yaml'
+                    '| grep profile'
+                )
+                self.assertEqual(result.return_code, 0)
+                # Runs the actual oscap scan on the vm/clients and
+                # uploads report to Internal Capsule.
+                vm.execute_foreman_scap_client()
+                # Assert whether oscap reports are uploaded to
+                # Satellite6.
+                self.assertTrue(self.oscapreports.search(host))
+                self.oscappolicy.update(
+                    name=policy_values.get('policy'),
+                    new_name=gen_string('alpha'),
+                    content=OSCAP_DEFAULT_CONTENT['rhel_firefox'],
+                    profile=OSCAP_PROFILE['firefox'],
+                    period=OSCAP_PERIOD['weekly'],
+                    period_value=OSCAP_WEEKDAY['friday'],
+                )
+                self.oscapreports.delete(host, really=True)
+                for _ in range(2):
+                    vm.run(u'puppet agent -t 2> /dev/null')
+                updated_result = vm.run(
+                    u'cat /etc/foreman_scap_client/config.yaml'
+                    '| grep profile'
```

Comment:
As the test talks about updating oscap content, we need to check for "content" and not "profile". I know this should also work but, can you please check if we can instead grep for "content" and not "profile".